### PR TITLE
FISH-5725: adding code to support get Claims from ID Token

### PR DESCRIPTION
**Pull request by breakponchito**
Adding support to get User Claims from ID Token

### CODE REVIEW

File: security-connectors-api/src/main/java/fish/payara/security/annotations/OpenIdAuthenticationDefinition.java [hunk 1]
```diff
@@ -359,4 +366,10 @@
      * The Microprofile Config key for evaluating EL expressions for every HTTP session is <code>{@value}</code>
      */
     String OPENID_MP_SESSION_SCOPED_CONFIGURATION = "payara.security.openid.sessionScopedConfiguration";
+
+    /**
+     * The Microprofile Config key to skip the /userinfo endpoint call
+     * and get the user information from ID Token is <code>{@value}</code>
+     */
+    String OPENID_MP_CLAIMS_FROM_ID_TOKEN = "payara.security.openid.claimsFromIDToken";
```
Comment (pdudits) [inline security-connectors-api/src/main/java/fish/payara/security/annotations/OpenIdAuthenticationDefinition.java:374]: I suggest we are consistent with rest of code and call the property `userClaimsFromIdToken` -- we are not capitalizing abbreviations in other properties (URI is the only exception), and `user` is important distinction.

**Review by pdudits — APPROVED** [review]
At the beginning I was unsure about the name "userClaimsFromIDToken" but by the end I accepted it, and we should use it all around the code

File: openid/src/main/java/fish/payara/security/openid/domain/OpenIdContextImpl.java [hunk 2]
```diff
@@ -162,14 +164,41 @@ public void setExpiresIn(Long expiresIn) {
...
+     */
+    private JsonObject processClaimsFromIDToken() {
+        JwtClaims identityTokenJWTClaims = identityToken.getJwtClaims();
+        JwtClaims accessTokenJWTClaims = accessToken.getJwtClaims();
+        //setting profile claims from id token
+        JsonObject userInfo = Json.createObjectBuilder()
+                .add(OpenIdConstant.SUBJECT_IDENTIFIER, identityTokenJWTClaims.getStringClaim(OpenIdConstant.SUBJECT_IDENTIFIER).orElse(""))
+                .add(OpenIdConstant.NAME, identityTokenJWTClaims.getStringClaim(OpenIdConstant.NAME).orElse(""))
+                .add(OpenIdConstant.FAMILY_NAME, accessTokenJWTClaims.getStringClaim(OpenIdConstant.FAMILY_NAME).orElse(""))
+                .add(OpenIdConstant.GIVEN_NAME, accessTokenJWTClaims.getStringClaim(OpenIdConstant.FAMILY_NAME).orElse(""))
```
Comment (aubi) [inline openid/src/main/java/fish/payara/security/openid/domain/OpenIdContextImpl.java:192]: Probably a copy-error, you obviously meant GIVEN_NAME in getStringClaim

File: security-connectors-api/src/main/java/fish/payara/security/annotations/OpenIdAuthenticationDefinition.java (hunk 1 above)
Comment (breakponchito) [reply security-connectors-api/src/main/java/fish/payara/security/annotations/OpenIdAuthenticationDefinition.java]: I already fix it, thanks

File: openid/src/main/java/fish/payara/security/openid/domain/OpenIdContextImpl.java (hunk 2 above)
Comment (breakponchito) [reply openid/src/main/java/fish/payara/security/openid/domain/OpenIdContextImpl.java]: I already fix it, thanks

**Pull request merged**